src/core/toga/widgets/table.py

```diff
@@ -136,3 +140,62 @@ def on_select(self, handler):
         """
         self._on_select = wrapped_handler(self, handler)
         self._impl.set_on_select(self._on_select)
+
+    def add_column(self, heading, accessor=None):
+        """
+        Add a new column to the table
+
+        :param heading:     title of the column
+        :type heading:      ``string``
+        :param accessor:    accessor of this new column
+        :type heading:      ``string``
+        """
+
+        if not accessor:
+            accessor = to_accessor(heading)
+
+        if accessor in self._accessors:
+            raise ValueError('Accesor "{}" is already in use'.format(accessor))
+
+        self.headings.append(heading)
+        self._accessors.append(accessor)
+
+        self._impl.add_column(heading, accessor)
+
+    def remove_column(self, column):
+        """
+        Remove a table column
+
+        :param column:     accessor or position (>0)
+        :type column:      ``string``
+        :type column:      ``int``
+        """
+
+        if type(column) == str and column in self._accessors:
+            # delete by accessor
+            accessor = column
+        elif type(column) == int and \
+                column >= 0 and column <= len(self._accessors)-1:
+            # delete by position
+            accessor = self._accessors[column]
+        else:
+            raise ValueError('Invalid column: "{}"'.format(column))
+
+        # remove column
+        try:
+            self._impl.remove_column(accessor)
+        except Exception as exception:
+            raise exception
+        else:
+            del self.headings[self._accessors.index(accessor)]
+            self._accessors.remove(accessor)
+
+    def get_missing_value(self, numrow, accessor):
```

Comment:
This is an "internal" API - it's not something that end-users should ever call. At the very least, this should be prefixed with an underscore to indicate that it's not an API for public use. I'm also not wild about the use of the "get" prefix - that's a very "Java-esque" convention; although in this case I don't know if there's a better option. 

I'd be tempted to turn this into a `missing_value` property that raises a ValueError if `_missing_value` is None, with the message of the exception being a format string; and then, back in the implementation where this value is being used, apply the format string with the numrow/accessor context. i.e.,
```
@property
def missing_value(self):
    if self._missing_value is None:
        raise ValueError("Row '{}' doesn't support accessor '{}'.... ", '')
    return self._missing_value
```
and then when it is used:
```
try:
    value = self.interface.missing_value
except ValueError as e:
    message, value = e.args
    print(message.format(row, col_identifier))
```

Reply:
I changed to a property I had to put the last try/except nested which doesn't look great. I didn't know how to solve avoiding it.